content/articles/building-scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/index.md

```diff
@@ -0,0 +1,115 @@
+![Continue Project](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/springstart1.jpg)
+
+Now, we can see the project open and the dependencies we selected, displayed in the `pom.xml` file.
+![Pom File](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/parentpom.jpg)
+![Landing Page](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/landing.jpg)
+
+Time for creating modules. First the data module. This module houses the entities for the project and the interfaces (JPA Repositories) that allow us interact seamlessly with out database, MySQL in this case. To create the first module, right click on the project name `demohospital` and click `New` > `Module`.
+On the resulting modal form, select `Maven` and click `Next`.
+
+![Setting up](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/settingup.jpg)
+
+Type in a Name (this will be the name of the module – in this case, `data`), leave `Location` as is and click `Finish`. Spring Boot does the rest for you.
+
+![Data](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/data.jpg)
+
+A `data` module has been created by you.
+
+![Data Two](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/datatwo.jpg)
+
+Following the same approach, let us create the `Service` and `Controller` modules. Right-click on the project name, click `New` > `Module`. Select `Maven`, type in a Name (this time, it would be `Service`), select `Finish`. You now have a `Service` module.
+
+![Service](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/service.jpg)
+
+The exact same approach for the `controller` module, as shown in the picture below
+
+![Controller](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/controller.jpg)
+
+As it is now, the services needed by the hospital can be created in each of these modueles and written to interact with one another.
+
+### How can we make them interact with each other?
+As mentioned earlier, the data module holds the models and database operations for the entities, the services houses the implementations of the entities and the controller is the entry point into the project, where the endpoints are created. To make the methods of a module available for another (for instance, to make `data` module methods available to `service` module), simply paste the following in the `pom.xml` file of the `service` module and install it:
+
+```java
```

Comment:
Is it not xml?

Reply:
Adjusted accordingly. I actually did not know about that tag until now. Thanks